extensions/src/org/pentaho/platform/osgi/KarafInstance.java

```diff
@@ -139,7 +136,7 @@ private void assignInstanceNumber() {
     }
 
     instanceNumber = testInstance;
-
+    System.setProperty( "karaf.data", cachePath );
```

Comment:
If the karaf.data property is already set, we should not override its value.
Unless we really need to.
Deferring to @pentaho-nbaker or @rmansoor on whether this is ok or if we should call fillMissedSystemProperty instead